src/vignette/util/thumbnail.clj

```diff
@@ -109,10 +110,13 @@
              :response-code 404}
             "unable to get original for thumbnailing")))
 
+(defn- extract-extension [filename]
+  (last (split filename #"\.")))
+
 (defn original->local
   "Take the original and make it local."
   [original]
-  (let [temp-file (io/file (temp-filename))]
+  (let [temp-file (io/file (temp-filename nil (extract-extension (filename original))))]
```

Comment:
Ah, ok, this looks like the part that fixes the previous PR. Can you add tests for this behavior?